components/CategoryToNumber.js

```diff
@@ -7,14 +7,26 @@ exports.getComponent = function () {
   c.inPorts.add('in', {
     datatype: 'string',
   });
+  c.inPorts.add('dataset', {
+    datatype: 'string',
+    control: true,
+  });
   c.outPorts.add('out', {
     datatype: 'number',
   });
   c.process((input, output) => {
-    if (!input.hasData('in')) {
+    if (!input.hasData('in', 'dataset')) {
       return;
     }
     const data = input.getData('in');
+    const dataset = input.getData('dataset');
+    if (dataset !== 'parkspaces') {
```

Comment:
Hiding this logic inside the component when it needs specific info from outside to tell it how to behave seems unnecessary obscure?